### PR TITLE
Add UTs for ExtractResourceRestrictions

### PR DESCRIPTION
**Pull request by nessiLahav**
### What does this PR do?
Add UTs for ExtractResourceRestrictions

#### Change log
- [ ] The CHANGELOG has been updated, or
- [ ] This PR does not include user-facing changes and doesn't require a CHANGELOG update

#### Test coverage
- [ ] This PR includes new unit and integration tests to go with the code changes, or
- [ ] The changes in this PR do not require tests

#### Documentation
- [ ] Docs (e.g. `README`s) were updated in this PR, and/or there is a follow-on issue to update docs, or
- [ ] This PR does not require updating any documentation

#### API Changes
- [ ] The [OpenAPI spec](https://github.com/cyberark/conjur-openapi-spec) has been updated to meet new API changes (or an issue has been opened), or
- [ ] The changes in this PR do not affect the Conjur API

### CODE REVIEW

File: spec/app/domain/authentication/resource_restrictions/extract_resource_restrictions_spec.rb [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    )
+
+    allow(mocked_fetch_resource_annotations_instance_empty).to(
+      receive(:call).and_return({})
+    )
+
+    allow(mocked_fetch_resource_annotations_instance_valid).to(
+      receive(:call).and_return(
+        "#{authenticator_name}/#{service_id}/claim1" => "claim1-value",
+        "#{authenticator_name}/#{service_id}/claim2" => "claim2-value"
```
Comment (sashaCher) [inline spec/app/domain/authentication/resource_restrictions/extract_resource_restrictions_spec.rb:35]: Do you want to add annotation belongs to other service id and to validate that it's ignored
And to add not related annotation at all

Comment (nessiLahav) [reply spec/app/domain/authentication/resource_restrictions/extract_resource_restrictions_spec.rb:35]: @sashaCher This is the responsibility of this class `GetServiceSpecificRestrictionFromAnnotation` and we already have UT for it

Comment (sashaCher) [reply spec/app/domain/authentication/resource_restrictions/extract_resource_restrictions_spec.rb:35]: From one side you're right.
From other you do not mock it so the tests are running the original class behavior and this is the place where UT starting to look a bit more "integrationnal". Means we can catch if the way we're integrating classes together is right.
CC: @eladkug

**Review by sashaCher — APPROVED** [review]
LGTM

**Pull request merged**